android/src/main/java/com/polidea/flutter_ble_lib/converter/ScanResultJsonConverter.java

```diff
@@ -0,0 +1,92 @@
+package com.polidea.flutter_ble_lib.converter;
+
+import android.support.annotation.NonNull;
+
+import com.polidea.multiplatformbleadapter.AdvertisementData;
+import com.polidea.multiplatformbleadapter.ScanResult;
+import com.polidea.multiplatformbleadapter.utils.Base64Converter;
+import com.polidea.multiplatformbleadapter.utils.UUIDConverter;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class ScanResultJsonConverter implements JsonConverter<ScanResult> {
+
+    private interface Metadata {
+        String ID = "id";
+        String NAME = "name";
+        String RSSI = "rssi";
+        String MTU = "mtu";
+
+        String MANUFACTURER_DATA = "manufacturerData";
+        String SERVICE_DATA = "serviceData";
+        String SERVICE_UUIDS = "serviceUUIDs";
+        String LOCAL_NAME = "localName";
+        String TX_POWER_LEVEL = "txPowerLevel";
+        String SOLICITED_SERVICE_UUIDS = "solicitedServiceUUIDs";
+        String IS_CONNECTABLE = "isConnectable";
+        String OVERFLOW_SERVICE_UUIDS = "overflowServiceUUIDs";
+    }
+
+    @Override
+    public String toJson(ScanResult value) throws JSONException {
+        JSONObject root = new JSONObject();
+        root.put(Metadata.ID, value.getDeviceId() != null ? value.getDeviceId() : JSONObject.NULL);
+        root.put(Metadata.NAME, value.getDeviceName() != null ? value.getDeviceName() : JSONObject.NULL);
+        root.put(Metadata.RSSI, value.getRssi());
+        root.put(Metadata.MTU, value.getMtu());
+        if (value.getAdvertisementData() != null) {
+            serializeAdvertisementData(root, value.getAdvertisementData());
+        }
+        root.put(Metadata.IS_CONNECTABLE, JSONObject.NULL);
+        root.put(Metadata.OVERFLOW_SERVICE_UUIDS, JSONObject.NULL);
+        return root.toString();
+    }
+
+    private void serializeAdvertisementData(JSONObject root,
```

Comment:
what about extracting a few smaller private methods for sake of readability?

Reply:
Done.